src/Transport/Seam/MessagePump.cs

```diff
@@ -42,12 +42,7 @@ public Task Init(Func<PushContext, Task> pump, CriticalError criticalError, Push
 
                 circuitBreaker.Success();
 
-                var context = new ContextBag();
-
-                context.Set(receiveContext);
-
-                //todo, figure out what the TransportTransaction parameter is about
-                return messagePump(new PushContext(incoming.MessageId, incoming.Headers, incoming.BodyStream, new NoTransaction(), tokenSource, context));
+                return messagePump(new PushContext(incoming.MessageId, incoming.Headers, incoming.BodyStream, new ReceiveContextAwareTransaction(receiveContext), tokenSource, new ContextBag()));
```

Comment:
There's a some sort of mixup here... the `ContextBag` is empty, the receive context is sent via transaction. So the [test](https://github.com/Particular/NServiceBus.AzureServiceBus/blob/7032515acfa946a11dbb869564ebcf287162c115/src/Tests/Seam/When_dispatching_messages_in_receive_context.cs#L65) will throw an exception and abandon the message, causing an infinite loop.